akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -306,15 +306,31 @@ class Http2ServerSpec extends AkkaSpec with WithInPendingUntilFixed {
     }
 
     "respect the substream state machine" should {
-      "reject other frame than HEADERS/PUSH_PROMISE in idle state with connection-level PROTOCOL_ERROR (5.1)" in pending
+      abstract class SimpleRequestResponseRoundtripSetup extends TestSetup with RequestResponseProbes
+
+      "reject other frame than HEADERS/PUSH_PROMISE in idle state with connection-level PROTOCOL_ERROR (5.1)" in new SimpleRequestResponseRoundtripSetup {
+        sendDATA(9, endStream = true, HPackSpecExamples.C41FirstRequestWithHuffman)
+        expectGOAWAY(0)
```

Comment:
yeah I probably will handle that aswell, well my first idea was just completionStage the demuxer, also there are cases where you don't have a GOAWAY frame and still want to close the connection, i.e. malicious clients.